### ISSUE TITLE
Legend outside graph window

### ISSUE THREAD

**Issue by josephmure** [openturns/openturns#1688]
Is there a way to move the legend outside the graph window?

```
import openturns as ot
from openturns import viewer

irregular_str = 'exp(x1)/5 - x2/5 + x2^6/3 + 4*x2^4 - 4*x2^2 + 7*x1^2/10 + x1^4 + 3/(4*x1^2 + 4*x2^2 + 1)'
irregular = ot.SymbolicFunction(['x1', 'x2'], [irregular_str])

graph = irregular.draw([-1.0, -1.0], [1.0, 1.0])
graph.setLegendPosition('bottomright')
viewer.View(graph)
```
![irregular_function](https://user-images.githubusercontent.com/31989332/100877366-f553ec00-34a8-11eb-82dc-a0cdcd3822d9.png)

In the script above, I would like to be able to specify
```
graph.setLegendPosition('outsideright')
```
or something in order to move the legend out of the window, where it masks part of the graph. Is there some other method that can do it?

**Comment by jschueller** [comment]
I think you can do it using the legend_kw argument

**Comment by josephmure** [comment]
Thanks @jschueller, that works. I only had to change the last line of the script to
```
viewer.View(graph, legend_kw={'bbox_to_anchor':(1,1), 'loc':"upper left"})
```
to produce
![irregular_function](https://user-images.githubusercontent.com/31989332/101052883-6d90df00-3587-11eb-9505-44a2c7bf9854.png)

Still on the subject of graphs, I had another question. What is the difference between the "square"/"fsquare" point styles? Or between "circle"/"fcircle". I get the same output with both alternatives. The following part of my script follows what I submitted in my original post.

```
train_size = 3
test_size = 3
ot.RandomGenerator.SetSeed(0)
uniform = ot.ComposedDistribution([ot.Uniform(-1.0, 1.0), ot.Uniform(-1.0, 1.0)])
initialTrainXSet = uniform.getSample(train_size)
initialTestXSet = uniform.getSample(test_size)

contour_legends = list(graph.getLegends())
for num, label in enumerate(contour_legends):
    contour_legends[num] = ''
graph.add(ot.Cloud(initialTrainXSet, 'blue', 'fsquare')) # I assume "fsquare" means "full square".
contour_legends.append('Train')
graph.add(ot.Cloud(initialTestXSet, 'red', 'square')) # But "square" is no different!
contour_legends.append('Test')
graph.setLegends(contour_legends)
viewer.View(graph)
```
![irregular_function](https://user-images.githubusercontent.com/31989332/101053436-13dce480-3588-11eb-9fc2-36a5ca4a5f6f.png)

**Comment by regislebrun** [comment]
@josephmure Initially we were relying on R to implement the draw() method of graphs, and within R these point styles are different. Unfortunately, matplotlib does not allow to set the color of the marker outline, so these point styles produce the same marker.

**Comment by sofianehaddad** [comment]
hmmm I think matplotlib does the job, no ?
Having a look at R output
![image](https://user-images.githubusercontent.com/1433220/101085454-bc526f00-35af-11eb-9871-f8a3330952d7.png)


we can have 'empty' squares using matplotlib:

```
import matplotlib.pyplot as plt
import numpy as np
plt.plot(np.random.randn(10), '.', color='b', marker='s',  mfc='none')
```
which produces:
![image](https://user-images.githubusercontent.com/1433220/101085508-cffdd580-35af-11eb-8dd1-8334d01a714b.png)

**Comment by regislebrun** [comment]
So the View class should be updated.

**Comment by josephmure** [comment]
Follow-up question: how do you set legend kwargs when your graph is inside a `GridLayout`?

**Comment by jschueller** [comment]
legend_kw is not passed to subplots

**Comment by sofianehaddad** [comment]
@jschueller I think we should fix the separation between `square`/`fsquare`, `circle`/`fcircle`
I can try.
If we export the `graph` above using `R` procedures, we get something like that:
![image](https://user-images.githubusercontent.com/1433220/107790118-b799fc00-6d52-11eb-98f4-02e4481812a2.png)

But I am afraid if we use `square` or circle as something not `full` we create confusion for users. What do you think?

**Comment by jschueller** [comment]
Yes, I dont think its too bad if we start to remove R keywords or such if it makes things easier for the viewer.

**Comment by sofianehaddad** [comment]
the thing is that `R square` does not match with `matplotlib square`
What do you propose?

**Comment by josephmure** [comment]
If it is a choice between the matplotlib convention and the R convention, I would go with matplotib. After all, nobody expects a Python module to maintain R conventions.

**Comment by sofianehaddad** [comment]
hmm are you relying on `master`?
Have a look at #1773

**Comment by josephmure** [comment]
Indeed, your PR fixes this issue. Thanks!

**Comment by mbaudin47** [comment]
Should this issue be closed?

**Comment by jschueller** [comment]
yes
